Challenge1/src/agent.spec.ts

```diff
@@ -1,104 +1,147 @@
+const mockOtherAddress = createAddress("0x03");
+
+const ifaceForSetData = new ethers.utils.Interface([CREATE_AGENT_ABI, UPDATE_AGENT_ABI]);
+const encodedCreateAgentData = ifaceForSetData.encodeFunctionData("createAgent", [
+  MOCK_AGENT_ID,
+  mockNethermindAddress,
+  "metadata",
+  MOCK_CHAIN_IDS,
+]);
+const encodedUpdateAgentData = ifaceForSetData.encodeFunctionData("updateAgent", [
+  MOCK_AGENT_ID,
+  "metadata",
+  MOCK_CHAIN_IDS,
+]);
+
+describe("Detection of bot deployments and updates, only by Nethermind, only to Forta Registry.", () => {
+  let handleTransaction: HandleTransaction;
+  let mockTxEvent = new TestTransactionEvent();
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(
+      CREATE_AGENT_ABI,
+      UPDATE_AGENT_ABI,
+      mockNethermindAddress,
+      mockFortaRegistryAddress
+    );
+  });
 
-let handleTransaction: HandleTransaction;
-const mockNetherMindAddress = "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8";
-const mockFortaRegistryAddress = "0x61447385B019187daa48e91c55c02AF1F1f3F863";
+  beforeEach(() => {
+    mockTxEvent = new TestTransactionEvent();
+  });
 
-beforeAll(() => {
-  handleTransaction = agent.handleTransaction;
-});
+  it("1. returns empty findings if not from Nethermind address", async () => {
+    mockTxEvent.setFrom(mockOtherAddress).setTo(mockFortaRegistryAddress).setData(encodedCreateAgentData);
 
-describe("returns empty findings if transaction is not from Nethermind address", () => {
-  it("should return empty findings", async () => {
-    const mockTxEvent = {
-      from: "0x1234567890123456789012345678901234567890",
-      to: mockFortaRegistryAddress,
-      filterFunction: jest.fn().mockReturnValue([]),
-    } as unknown as TransactionEvent;
+    // .addTraces({
+    //   from: mockNethermindAddress,
+    //   to: mockFortaRegistryAddress,
+    //   function: CREATE_AGENT_ABI,
+    //   arguments: [MOCK_AGENT_ID, mockNethermindAddress, "metadata", MOCK_CHAIN_IDS],
+    // });
 
     const findings = await handleTransaction(mockTxEvent);
-
     expect(findings).toHaveLength(0);
   });
-});
 
-describe("returns empty findings if transaction is not to Forta address", () => {
-  it("should return empty findings", async () => {
-    const mockTxEvent = {
-      from: mockNetherMindAddress,
-      to: "0x1234567890123456789012345678901234567890",
-      filterFunction: jest.fn().mockReturnValue([]),
-    } as unknown as TransactionEvent;
+  it("2. returns empty findings if not to Forta Registry", async () => {
+    mockTxEvent.setFrom(mockNethermindAddress).setTo(mockOtherAddress).setData(encodedCreateAgentData);
+
+    // .addTraces({
+    //   from: mockNethermindAddress,
+    //   to: mockFortaRegistryAddress,
+    //   function: CREATE_AGENT_ABI,
+    //   arguments: [MOCK_AGENT_ID, mockNethermindAddress, "metadata", MOCK_CHAIN_IDS],
+    // });
 
     const findings = await handleTransaction(mockTxEvent);
+    expect(findings).toHaveLength(0);
```

Comment:
```suggestion
    expect(findings).toStrictEqual([]);
```
Team opts to use `.toStrictEqual()`, for both lack of findings and for expected findings. Applicable elsewhere as well.